src/Pagination/index.js

```diff
@@ -0,0 +1,195 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+
+import { Icon, Theme } from '..';
+
+import {
+  Container,
+  PaginationContainer,
+  PaginationItem,
+  PaginationGroup,
+  ArrowContainer,
+  BreakItem,
+} from './elements';
+
+/**
+ * Pagination
+ *
+ * This component is in charge of displaying
+ * a Pagination
+ *
+ * @param {number} pageCount - total number of pages in the pagination.
+ * @param {number} pageRangeDisplayed - displayed page range.
+ * @param {number} marginPageDisplayed - page displayed on the pagination marges.
+ * @param {string} breakLabel - default '...'. Label when you need a break for long pagination.
+ * @param {func} getRequestedPageNumber - component callback function.
+ * @param {string} align - component alignment. oneOf ['left', 'center', 'right'].
+ *
+ * @return {jsx}
+ */
+const Pagination = ({
+  pageCount,
+  pageRangeDisplayed,
+  marginPageDisplayed,
+  breakLabel,
+  getRequestedPageNumber,
+  align,
+}) => {
+  /**
+   * Component initial state.
+   */
+  const [currentPage, setCurrentPage] = useState(1);
+
+  /**
+   * Page item click callback.
+   *
+   * @param {number} index
+   */
+  const handlePageItemClicked = index => {
+    if (currentPage === index) return;
+
+    setCurrentPage(index);
+    getRequestedPageNumber(currentPage);
+  };
+
+  /**
+   * Handle click on previous page.
+   */
+  const handlePrevious = () => {
+    setCurrentPage(currentPage === 1 ? 1 : currentPage - 1);
+    getRequestedPageNumber(currentPage);
+  };
+
+  /**
+   * Handle click on next page.
+   */
+  const handleNextPage = () => {
+    setCurrentPage(currentPage === pageCount ? pageCount : currentPage + 1);
+    getRequestedPageNumber(currentPage);
+  };
+
+  /**
+   * Return a pagination item.
+   *
+   * @param {number} index
+   *
+   * @return {jsx}
+   */
+  const getPaginationItem = index => {
+    return (
+      <PaginationItem
+        key={index}
+        active={currentPage === index}
+        onClick={() => handlePageItemClicked(index)}
+        data-testid={`page-${index}`}
+      >
+        {index}
+      </PaginationItem>
+    );
+  };
+
+  /**
+   * Render the pagination.
+   *
+   * @return {Array}
+   */
+  const renderPagination = () => {
+    const items = [];
+
+    if (pageCount <= pageRangeDisplayed) {
+      for (let i = 1; i <= pageCount; i++) {
+        items.push(getPaginationItem(i));
+      }
+    } else {
+      let left = pageRangeDisplayed / 2;
+      let right = pageRangeDisplayed - left;
+      let pageItem = index => getPaginationItem(index);
```

Comment:
Little question, why did you use `pageItem` here instead of using directly  `getPaginationItem` ?

Reply:
@Artikodin good question, there is no valid answer so I'll remove this useless `let` :)

Reply:
cool :)